### PR TITLE
[Console] Fix clearing sections containing questions

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 4.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28615 
| License       | MIT
| Doc PR        | n/a

Given:
```php
$section = $output->section();
$question = "What's your name?";
$name = (new QuestionHelper())->ask($input, $section, new Question($question));
$section->clear();
$section->writeln("$question (<info>$name</info>)");
```

Before:

After:

### CODE REVIEW

File: src/Symfony/Component/Console/Helper/QuestionHelper.php [hunk 1]
```diff
@@ -123,6 +124,10 @@ private function doAsk(OutputInterface $output, Question $question)
             $ret = trim($this->autocomplete($output, $question, $inputStream, \is_array($autocomplete) ? $autocomplete : iterator_to_array($autocomplete, false)));
         }
 
+        if ($output instanceof ConsoleSectionOutput) {
+            $output->addContent($ret);
```
Comment (pierredup) [inline src/Symfony/Component/Console/Helper/QuestionHelper.php]: How will this work with hidden input? There might not be any additional lines to clear in that case

Comment (chalasr) [reply src/Symfony/Component/Console/Helper/QuestionHelper.php]: Answering questions still implies to hit `enter`, empty and hidden inputs result in a blank line on the screen. Just tried it, works as expected with hidden questions

**Review by pierredup — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
(4.1 or master? see target branch vs description)

**Comment by chalasr** [comment]
@nicolas-grekas 4.1, fixed thanks

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Console/Tests/Output/ConsoleSectionOutputTest.php [hunk 2]
```diff
@@ -13,17 +13,20 @@
...
+use Symfony\Component\Console\Question\Question;
 
 class ConsoleSectionOutputTest extends TestCase
 {
     private $stream;
 
     protected function setUp()
     {
-        $this->stream = fopen('php://memory', 'r+', false);
+        $this->stream = fopen('php://memory', 'r+b', false);
```
Comment (fabpot) [inline src/Symfony/Component/Console/Tests/Output/ConsoleSectionOutputTest.php]: should be reverted /cc @nicolas-grekas

File: src/Symfony/Component/Console/Tests/Output/ConsoleSectionOutputTest.php [hunk 3]
```diff
@@ -137,4 +140,24 @@ public function testMultipleSectionsOutput()
         rewind($output->getStream());
         $this->assertEquals('Foo'.PHP_EOL.'Bar'.PHP_EOL."\x1b[2A\x1b[0JBar".PHP_EOL."\x1b[1A\x1b[0JBaz".PHP_EOL.'Bar'.PHP_EOL."\x1b[1A\x1b[0JFoobar".PHP_EOL, stream_get_contents($output->getStream()));
     }
+
+    public function testClearSectionContainingQuestion()
+    {
+        $inputStream = fopen('php://memory', 'r+b', false);
```
Comment (fabpot) [inline src/Symfony/Component/Console/Tests/Output/ConsoleSectionOutputTest.php]: same here, no `b` needed

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**